test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -510,9 +510,195 @@ bool VerifyShowToast_Unpackaged()
+        if (removeToastAsync.wait_for(std::chrono::seconds(300)) == winrt::Windows::Foundation::AsyncStatus::Error)
+        {
+            removeToastAsync.GetResults();
+        }
+
+        removeToastAsync.Cancel();
+        return false;
+    }
+    catch (...)
+    {
+        return winrt::to_hresult() == E_INVALIDARG;
+    }
+}
+
+bool VerifyFailedRemoveWithIdentiferAsyncUsingNonActiveToastIdentifier()
+{
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+
+    winrt::ToastNotification toast{ CreateToastNotification() };
+    toastNotificationManager.ShowToast(toast);
+    auto id = toast.ToastId();
+    toastNotificationManager.RemoveAllAsync().get();
+
+    try
+    {
+        auto removeToastAsync{ winrt::ToastNotificationManager::Default().RemoveWithIdentiferAsync(id) };
+        if (removeToastAsync.wait_for(std::chrono::seconds(300)) == winrt::Windows::Foundation::AsyncStatus::Error)
+        {
+            removeToastAsync.GetResults();
+        }
+
+        removeToastAsync.Cancel();
+        return false;
+    }
+    catch (...)
+    {
+        return winrt::to_hresult() == E_FAIL; // ELx - would expect the operation to report that it couldn't remove the toast but it doesn't. The operation simply fails silently. Is this a bug or the expected behavior?
+    }
+}
+
+bool VerifyRemoveWithIdentiferAsync()
+{
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+
+    winrt::ToastNotification toast1{ CreateToastNotification(L"Toast1") };
+    toastNotificationManager.ShowToast(toast1);
+
+    winrt::ToastNotification toast2{ CreateToastNotification(L"Toast2") };
+    toastNotificationManager.ShowToast(toast2);
+
+    winrt::ToastNotification toast3{ CreateToastNotification(L"Toast3") };
+    toastNotificationManager.ShowToast(toast3);
+
+    if (!VerifyToastIsActive(toast1.ToastId()) || !VerifyToastIsActive(toast2.ToastId()) || !VerifyToastIsActive(toast3.ToastId()))
+    {
+        return false;
+    }
+
+    toastNotificationManager.RemoveWithIdentiferAsync(toast2.ToastId()).get();
+
+    if (!VerifyToastIsActive(toast1.ToastId()) || !VerifyToastIsActive(toast3.ToastId()))
+    {
+        return false;
+    }
+
+    return !VerifyToastIsActive(toast2.ToastId());
+}
+
+bool VerifyRemoveWithTagAsync()
+{
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+
+    winrt::ToastNotification toast1{ CreateToastNotification() };
+    toast1.Tag(L"Shared tag");
+    toast1.Group(L"group1");
+    toastNotificationManager.ShowToast(toast1);
+
+    winrt::ToastNotification toast2{ CreateToastNotification() };
+    toast2.Tag(L"Shared tag");
+    toast2.Group(L"group2");
+    toastNotificationManager.ShowToast(toast2);
+
+    winrt::ToastNotification toast3{ CreateToastNotification() };
+    toast3.Tag(L"Shared tag");
+    toast3.Group(L"group3");
+    toastNotificationManager.ShowToast(toast3);
+
```

Comment:
looks like all 3 test toasts have the same tag, maybe we can add one more toast with a different tag and verify that it will not be removed.